app/views/job_invocations/show.json.erb

```diff
@@ -0,0 +1,4 @@
+{
+  "autoRefresh": "<%= @auto_refresh %>",
+  "hosts": <%= targeting_hosts(@job_invocation, @hosts).to_json.html_safe %>
+}
```

Comment:
I think this could be moved to the controller where you can use 
```rb
render json: {
        autoRefresh: ...
        hosts: ...,
      }, status: :ok
```
and than in the API call from the UI, you don't need to set `format=json` in the url

Reply:
yeah, but `JobInvocationController#show` is shared for two actions: GET html view for user and API call for getting data for this component. So I'd have to do something like this:
```
respond_to do |format|
  format.html do
    ...
  end
   format.text do
    render :json ....
   end
end
``` 
which is IMHO basically same as the how I did it - just add `show.json.erb` and have `format` param in URL.